src/astgen.zig

```diff
@@ -2441,10 +2457,116 @@ fn nodeMayNeedMemoryLocation(start_node: *ast.Node) bool {
             .For,
             .Switch,
             .Call,
-            .BuiltinCall, // TODO some of these can return false
             .LabeledBlock,
             => return true,
 
+            .BuiltinCall => {
+                const name = scope.tree().tokenSlice(node.castTag(.BuiltinCall).?.builtin_token);
+                if (mem.eql(u8, name, "@addWithOverflow")) return true;
```

Comment:
Use `std.ComptimeStringMap(bool)` here unless that is causing some issues with comptime.